app/Domains/Settings/ManageTemplates/Services/UpdateModulePosition.php

```diff
@@ -89,29 +87,27 @@ private function updatePosition(): void
             $this->updateDescendingPosition();
         }
 
-        DB::table('module_template_page')
-            ->where('template_page_id', $this->templatePage->id)
-            ->where('module_id', $this->module->id)
+        $this->module->pivot
             ->update([
                 'position' => $this->data['new_position'],
             ]);
     }
 
     private function updateAscendingPosition(): void
     {
-        DB::table('module_template_page')
-            ->where('template_page_id', $this->templatePage->id)
-            ->where('position', '>', $this->pastPosition)
-            ->where('position', '<=', $this->data['new_position'])
+        $this->templatePage->modules()
+            ->wherePivot('position', '>', $this->pastPosition)
+            ->wherePivot('position', '<=', $this->data['new_position'])
+            ->newPivotQuery()
             ->decrement('position');
```

Comment:
@djaiss This one is tricky: we need to update the positon's pivot value. Thanks to `newPivotQuery` we can do that.